lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -140,81 +183,66 @@ def test__has_access_descriptor(self):
         with self.assertRaises(ValueError):
             access._has_access_descriptor(user, 'not_load_or_staff', descriptor)
 
+    @ddt.data(
+        (True, None, False, access_response.VisibilityError),
+        (False, None, True),
+        (True, YESTERDAY, False, access_response.VisibilityError),
+        (False, YESTERDAY, True),
+        (True, TOMORROW, False, access_response.VisibilityError),
+        (False, TOMORROW, False, access_response.StartDateError)
+    )
+    @ddt.unpack
     @patch.dict('django.conf.settings.FEATURES', {'DISABLE_START_DATES': False})
-    def test__has_access_descriptor_staff_lock(self):
+    def test__has_access_descriptor_staff_lock(self, staff_lock, start, expected_access, expected_error_type=None):
```

Comment:
The `expected_access` should be derived from `expected_error_type` so it's not passed in through the `ddt`.  It will also make the `ddt` parameters easier to read.